react-validated-input-component/my-app/src/ValidatedInput.tsx

```diff
@@ -0,0 +1,64 @@
+import { FormEvent, useState } from 'react';
+import { FaCheck } from 'react-icons/fa';
+import { FaTimes } from 'react-icons/fa';
+import './ValidatedInput.css';
+
+export function ValidatedInput() {
+  const [password, setPassword] = useState('');
+  let psIsValid = false;
+  validatePassword();
+  function handleSubmit(e: FormEvent) {
+    e.preventDefault();
+  }
+
+  function validatePassword() {
+    if (password.length >= 8) {
+      psIsValid = true;
+    } else {
+      psIsValid = false;
+    }
+  }
+
+  function ValidOrNot() {
```

Comment:
React documentation says that you should not have nested components. That is, every component function must be at the top level. Most of the time it works, but occasionally it can cause problems. So it's best to follow their advice and never use nested components.